job/server/src/main/java/alluxio/master/job/plan/PlanCoordinator.java

```diff
@@ -194,9 +193,9 @@ public long getJobId() {
   public void setJobAsFailed(String errorType, String errorMessage) {
     synchronized (mPlanInfo) {
       if (!mPlanInfo.getStatus().isFinished()) {
-        mPlanInfo.setStatus(Status.FAILED);
         mPlanInfo.setErrorType(errorType);
         mPlanInfo.setErrorMessage(errorMessage);
+        mPlanInfo.setStatus(Status.FAILED);
```

Comment:
- it's kinda tricky that it requires these setters to be called in certain order. The code will be hard to maintain in the future.
- shall we call setJobAsFailed here?

Reply:
the idea is all calls to set status as failed will go through this method, and therefore the order is enforced.